File: DI/DI-Dagger-Java/src/main/java/model/casa/Puerta.java
package model.casa;

import javax.inject.Inject;
import java.util.UUID;

//@Singleton // Aquñi no hace nada, pero provemos!!!
public class Puerta {
    private final String id = UUID.randomUUID().toString();

    @Inject
    public Puerta() {

    }

    public void abrir() {
        System.out.println("Abriendo puerta");
    }

    public void cerrar() {
        System.out.println("Cerrando puerta");
    }

    @Override
    public String toString() {
        return "Puerta{" +
                "id='" + id + '\'' +
                '}';
    }
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/models/casas/Puerta.kt
package models.casas

import java.util.*
import javax.inject.Inject

class Puerta
@Inject constructor() {
    val id: String = UUID.randomUUID().toString()
    private var isOpened: Boolean = false
    fun abrir() {
        println("Abriendo puerta")
        isOpened = true
    }

    fun cerrar() {
        println("Cerrando puerta")
        isOpened = false
    }

    override fun toString(): String {
        return "Puerta(id='$id', isOpened=$isOpened)"
    }
}